raiden/network/proxies/token_network.py

```diff
@@ -886,49 +934,53 @@ def _set_total_deposit(
         checking_block = self.client.get_checking_block()
         amount_to_deposit = TokenAmount(total_deposit - previous_total_deposit)
 
-        # If there are channels being set up concurrently either the
-        # allowance must be accumulated *or* the calls to `approve` and
-        # `setTotalDeposit` must be serialized. This is necessary otherwise
-        # the deposit will fail.
-        #
-        # Calls to approve and setTotalDeposit are serialized with the
-        # deposit_lock to avoid transaction failure, because with two
-        # concurrent deposits, we may have the transactions executed in the
-        # following order
-        #
-        # - approve
-        # - approve
-        # - setTotalDeposit
-        # - setTotalDeposit
-        #
-        # in which case the second `approve` will overwrite the first,
-        # and the first `setTotalDeposit` will consume the allowance,
-        # making the second deposit fail.
-        self.token.approve(allowed_address=Address(self.address), allowance=amount_to_deposit)
+        transaction_hash = None
+        with self.token.approve_transaction(
```

Comment:
The goal was to make this block take the least amount of time possible. Sadly I couldn't remove the transaction polling from here, since I did not find a way of reliably using estimate gas for the `set_total_deposit` afterwards. This PR does however remove the polling of the deposit itself, which halves the time which the `approve` lock is acquired.